src/main/java/sirius/web/templates/pdf/AsyncLoadedImageElement.java

```diff
@@ -62,20 +64,44 @@ public AsyncLoadedImageElement(PdfReplaceHandler handler, String uri, int cssWid
         ScopeInfo scopeInfo = UserContext.getCurrentScope();
         resolvingThread = Thread.startVirtualThread(() -> {
             UserContext.get().setCurrentScope(scopeInfo);
-            try {
-                semaphore.acquire();
-                image = handler.resolveUri(uri, null, cssWidth, cssHeight);
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-                Exceptions.handle(e);
-            } catch (Exception e) {
-                Exceptions.handle(e);
-            } finally {
-                semaphore.release();
-            }
+            startResolvingResource(uri, handler, semaphore, 1);
         });
     }
 
+    private void startResolvingResource(String uri, PdfReplaceHandler handler, Semaphore semaphore, int attempt) {
+        Exception exception = null;
+        try {
+            semaphore.acquire();
+            image = handler.resolveUri(uri, null, cssWidth, cssHeight);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            exception = e;
+        } catch (Exception e) {
+            exception = e;
+        } finally {
+            semaphore.release();
+            if (image == null || exception != null) {
+                retryOrFail(uri, handler, semaphore, attempt, exception);
+            }
+        }
+    }
+
+    private void retryOrFail(String uri, PdfReplaceHandler handler, Semaphore semaphore, int attempt, Exception exception) {
+        if (exception instanceof InterruptedException) {
+            // we should accept that the thread is interrupted and don't try again
+            Exceptions.handle(exception);
+            return;
+        }
+        if (attempt <= MAX_ATTEMPTS) {
```

Comment:
```suggestion
        if (attempt < MAX_ATTEMPTS) {
```
I think this is an off by one error where we actually try 4 times if my thinking is correct